src/main/java/medic/gateway/alert/SmsSender.java

```diff
@@ -124,12 +127,13 @@ private void sendSms_dummy(WoMessage m) {
 	}
 
 	private PendingIntent intentFor(String intentType, WoMessage m, int partIndex, int totalParts) {
-		Intent intent = new Intent(intentType);
+		Intent intent = new Intent(ctx, IntentProcessor.class);
+		intent.setAction(intentType);
 		intent.putExtra("id", m.id);
 		intent.putExtra("partIndex", partIndex);
 		intent.putExtra("totalParts", totalParts);
 
-		return PendingIntent.getBroadcast(ctx, UNUSED_REQUEST_CODE, intent, PendingIntent.FLAG_ONE_SHOT);
+		return PendingIntent.getBroadcast(ctx, m.hashCode(), intent, PendingIntent.FLAG_ONE_SHOT);
```

Comment:
WoMessage doesn't define a hashcode so this will use the Object default which hashes the memory address of the Object which is ok, but if the message is fetched from the database again will be a different hashcode. I think it'd be better to use the hashcode of the messages ID which is immutable, eg

```suggestion
		return PendingIntent.getBroadcast(ctx, m.id.hashCode(), intent, PendingIntent.FLAG_ONE_SHOT);
```

Reply:
Thanks! Done